kafka/producer_test.go

```diff
@@ -146,7 +154,8 @@ func TestNewProducerBasic(t *testing.T) {
 			var actual []apmqueue.Record
 			orderingKeyToPartitionM := make(map[string]int32)
 			client.AddConsumeTopics(namespacedTopic)
-			for i := 0; i < len(batch)*partitionCount && len(actual) < len(batch); i++ {
```

Comment:
Unless I'm missing something, that first condition seemed incorrect ; if we want to read all records we must ensure that `len(actual) < len(batch)`.